subsys/net/ip/tcp2.c

```diff
@@ -163,7 +163,7 @@ static char *tcp_endpoint_to_string(union tcp_endpoint *ep)
 	}
 #undef BUF_SIZE
 #undef NBUFS
-	return s;
+	return log_strdup(s);
```

Comment:
This cannot be done like this. The log_strdup() call needs to be in the LOG_DBG() call, it cannot be used separately.

Reply:
@jukkar

> The log_strdup() call needs to be in the LOG_DBG() call, it cannot be used separately.

There's no other use for these functions outside of LOG_*(), but, OK, fixed.